File: LATIHAN1DPBO2023/CPP/Mahasiswa.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//library yang digunakan
#include <iostream>
#include <string>

using namespace std;

//deklarasi class Mahasiswa
class Mahasiswa
{
	//atribut yang digunakan (diset private)
	private:
		string name;
		string nim;
		string major;
		string faculty;

	//method yang digunakan (diset public)
	public:
		//constructor
		Mahasiswa()
		{
			this->name = "";
			this->nim = "";
			this->major = "";
			this->faculty = "";
		}

		/*setter dan getter digunakan karena atribut yang ada pada class Mahasiswa bersifat private, 
		sehingga jika ingin diakses oleh class lain harus melalui method*/

		//method untuk set (mengisi/menginput) nama mahasiswa
		void setName(string name)
		{
			this->name = name;
		}

		//method untuk set nim mahasiswa
		void setNim(string nim)
		{
			this->nim = nim;
		}

		//method untuk set program studi
		void setMajor(string major)
		{
			this->major = major;
		}

		//method untuk set fakultas
		void setFaculty(string faculty)
		{
			this->faculty = faculty;
		}

		//method untuk mendapatkan value dari atribut name
		string getName()
		{
			return this->name;
		}

		//method untuk mendapatkan value dari atribut nim
		string getNim()
		{
			return this->nim;
		}

		//method untuk mendapatkan value dari atribut major
		string getMajor()
		{
			return this->major;
		}

		//method untuk mendapatkan value dari atribut faculty
		string getFaculty()
		{
			return this->faculty;
		}

		//destructor
		~Mahasiswa()
		{

		}
};

File: LATIHAN1DPBO2023/PHP/Mahasiswa.php
<!-- Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin. -->

<?php

//deklarasi class Mahasiswa
class Mahasiswa
{
	//atribut
	private $name = "";
	private $nim = "";
	private $major = "";
	private $faculty = "";
	private $image = "";

	//constructor
	public function __construct($name = "", $nim = "", $major = "", $faculty = "", $image = "")
	{
		$this->name = $name;
		$this->nim = $nim;
		$this->major = $major;
		$this->faculty = $faculty;
		$this->image = $image;
	}

	//method untuk set nama
	public function setName($name)
	{
		$this->name = $name;
	}

	//method untuk mendapatkan value dari nama
	public function getName()
	{
		return $this->name;
	}

	//method untuk set nim
	public function setNim($nim)
	{
		$this->nim = $nim;
	}

	//method untuk mendapatkan value dari nim
	public function getNim()
	{
		return $this->nim;
	}

	//method untuk set program studi
	public function setMajor($major)
	{
		$this->major = $major;
	}

	//method untuk mendapatkan value dari program studi
	public function getMajor()
	{
		return $this->major;
	}

	//method untuk set fakultas
	public function setFaculty($faculty)
	{
		$this->faculty = $faculty;
	}

	//method untuk mendapatkan value dari fakultas
	public function getFaculty()
	{
		return $this->faculty;
	}

	//method untuk set foto profile
	public function setImage($image)
	{
		$this->image = $image;
	}

	//method untuk mendapatkan value dari foto profile
	public function getImage()
	{
		return $this->image;
	}

	//destructor
	public function __destruct()
	{

	}
}

?>
